### PR TITLE
app-leechcraft: use Qt5, modernize the eclass and ebuilds.

### PR DESCRIPTION
**Pull request by 0xd34df00d**
Indeed, a huge one.

1. Moved everything to EAPI 6 and some other changes to modernize the ebuilds.
2. Moved to Qt5.
3. Dropped some obsolete stuff like checks whether gcc ≥ 4.6 is used.
4. Dropped mentions of the leechcraft project in favour of my own name and proxy-maint as a backup.

### CODE REVIEW

**Comment by 0xd34df00d** [comment]
@SoapGentoo :]

File: eclass/leechcraft.eclass [hunk 1]
```diff
@@ -19,29 +19,28 @@
 #
 # Thanks for original eclass to Andrian Nord <NightNord@niifaq.ru>.
 #
-# Only EAPI >3 supported
+# Only EAPI >4 supported
 
 case ${EAPI:-0} in
-	4|5) ;;
+	4|5|6) ;;
```
Comment (Moiman) [inline eclass/leechcraft.eclass]: Why not drop support for anything else than EAPI=6?

File: app-leechcraft/lc-aggregator/lc-aggregator-9999.ebuild [hunk 2]
```diff
@@ -1,32 +1,31 @@
...
 
 DEPEND="~app-leechcraft/lc-core-${PV}[postgres?,sqlite?]
-	dev-qt/qtwebkit:4"
+	dev-qt/qtcore:5
+	dev-qt/qtsql:5[sqlite?,postgres?,mysql?]
+	dev-qt/qtwebkit:5
+	dev-qt/qtnetwork:5
+	dev-qt/qtprintsupport:5
+	dev-qt/qtwidgets:5
+	dev-qt/qtxml:5"
```
Comment (a17r) [inline app-leechcraft/lc-aggregator/lc-aggregator-9999.ebuild]: Please keep dependencies sorted :)

File: app-leechcraft/lc-azoth/lc-azoth-9999.ebuild [hunk 3]
```diff
@@ -15,24 +15,41 @@ IUSE="debug doc astrality +acetamide +adiumstyles +autoidler +autopaste +birthda
...
+	lastseen? (
+		dev-qt/qtconcurrent:5
+	)
+	otroid? (
+		dev-qt/qtconcurrent:5
+	)
+	autoidler? (
+		dev-qt/qtx11extras:5
+	)
+	autoidler? ( x11-libs/libXScrnSaver )
```
Comment (a17r) [inline app-leechcraft/lc-azoth/lc-azoth-9999.ebuild]: Duplicate USE flag, also sorting please

File: app-leechcraft/lc-azoth/lc-azoth-9999.ebuild [hunk 4]
```diff
@@ -15,24 +15,41 @@ IUSE="debug doc astrality +acetamide +adiumstyles +autoidler +autopaste +birthda
...
+	woodpecker? ( dev-libs/kqoauth )
+	xmpp? (
+		>=net-libs/qxmpp-0.9.3[qt5]
+		media? ( >=net-libs/qxmpp-0.9.3[qt5,speex] )
+	)
+	xtazy? (
+		~app-leechcraft/lc-xtazy-${PV}
+	)
+	crypt? ( app-crypt/qca:2[qt5] )
+	sarin? ( net-libs/tox )
```
Comment (a17r) [inline app-leechcraft/lc-azoth/lc-azoth-9999.ebuild]: Another duplicate USE

File: app-leechcraft/lc-blasq/lc-blasq-9999.ebuild [hunk 5]
```diff
@@ -12,19 +12,28 @@ SLOT="0"
...
+	dev-qt/qtnetwork:5
+	dev-qt/qtdeclarative:5
+	rappor? ( dev-qt/qtxml:5 )
+	deathnote? (
+		dev-qt/qtxml:5
+		dev-qt/qtxmlpatterns:5
+	)
+	spegnersi? ( dev-qt/qtxml:5 )
+	vangog? ( dev-qt/qtxml:5 )
+	spegnersi? ( dev-libs/kqoauth )
```
Comment (a17r) [inline app-leechcraft/lc-blasq/lc-blasq-9999.ebuild]: duplicate USE

File: app-leechcraft/lc-kbswitch/lc-kbswitch-9999.ebuild [hunk 6]
```diff
@@ -1,17 +1,23 @@
...
+DEPEND="~app-leechcraft/lc-core-${PV}
+	dev-qt/qtnetwork:5
+	dev-qt/qtwidgets:5
+	dev-qt/qtx11extras:5
+	dev-qt/qtdeclarative:5[widgets]
+"
 RDEPEND="${DEPEND}
-	x11-apps/setxkbmap"
+	x11-apps/setxkbmap
+	quark? ( ~virtual/leechcraft-quark-sideprovider-${PV} )"
```
Comment (a17r) [inline app-leechcraft/lc-kbswitch/lc-kbswitch-9999.ebuild]: Is it that important that it needs an optional RDEPEND as compared to a pkg_postinst message?

File: app-leechcraft/lc-poshuku/lc-poshuku-9999.ebuild [hunk 7]
```diff
@@ -1,49 +1,52 @@
...
+inherit leechcraft
 
 DESCRIPTION="Poshuku, the full-featured web browser plugin for LeechCraft"
 
 SLOT="0"
 KEYWORDS=""
-IUSE="+autosearch debug +dcac +cleanweb +fatape +filescheme +fua +idn +keywords +onlinebookmarks
-		+pcre postgres qrd +sqlite wyfv"
+IUSE="+autosearch debug +dcac +cleanweb +fatape +filescheme +foc +fua +idn +keywords +onlinebookmarks
+	  postgres qrd +speeddial +sqlite webengineview +webkitview"
```
Comment (a17r) [inline app-leechcraft/lc-poshuku/lc-poshuku-9999.ebuild]: Please rather use `webengine` and `webkit`, flags that are already in use by other packages in the tree.

File: app-leechcraft/liblaretz/metadata.xml [hunk 8]
```diff
@@ -1,9 +1,13 @@
-<?xml version="1.0" encoding="UTF-8"?>
+<?xml version='1.0' encoding='UTF-8'?>
```
Comment (a17r) [inline app-leechcraft/liblaretz/metadata.xml]: that must have happened by mistake

File: eclass/leechcraft.eclass (hunk 1 above)
Comment (SoapGentoo) [inline eclass/leechcraft.eclass]: yes please, keep it modern :wink:

File: app-leechcraft/lc-azoth/lc-azoth-9999.ebuild [hunk 9]
```diff
@@ -43,49 +60,52 @@ RDEPEND="${COMMON_DEPEND}
 	)
 	crypt? ( app-crypt/qca:2[gpg] )
 	latex? (
-		virtual/imagemagick-tools
+		|| (
+			media-gfx/imagemagick
+			media-gfx/graphicsmagick[imagemagick]
+		)
```
Comment (SoapGentoo) [inline app-leechcraft/lc-azoth/lc-azoth-9999.ebuild]: why did you take away the virtual? It's either virtual or USE flags, but no naked `|| ( )` deps anymore

**Comment by SoapGentoo** [comment]
@0xd34df00d I had to do an "emergency" merge, as I hosed the tree by deleting the project prematurely. Please still do all the improvementes suggested by @a17r

**Pull request closed without merging**